### ISSUE TITLE
Dgraph got stuck after uncommited transaction

### ISSUE THREAD

**Issue by makitka2007** [dgraph-io/dgraph#3128]
run 1 zero and 1 alpha (v.1.0.13)
```
dgraph zero --replicas 3 --idx 10 --my 10.66.10.49:5080 -o 0
```
```
dgraph alpha --lru_mb 1024 --my 10.66.10.49:7080 -o 0 --zero 10.66.10.49:5080
```
then run mutation using HTTP interface but *without* `X-DGraph-CommitNow` header. so don't commit a transaction:
```
POST 10.66.10.49:8080/mutate
{
    set {
        _:node <p> "" .
    }
}
```
after that just wait for 10 minutes or a little bit more
then all queries on this node got stuck. for example this one:
```
POST 10.66.10.49:8080/query?ro=true
{
    q(func: has(p)) {
        count(uid)
    }
}
```

in the logs on alpha there are following messages:
```
I0311 14:15:59.739153   16283 groups.go:394] Serving tablet for: p
I0311 14:21:25.824778   16283 draft.go:925] Found 1 old transactions. Acting to abort them.
I0311 14:21:25.827384   16283 draft.go:885] TryAbort 1 txns with start ts. Error: <nil>
I0311 14:21:25.827397   16283 draft.go:892] TryAbort returned with delta: txns:<start_ts:1 >
I0311 14:21:25.827450   16283 draft.go:909] TryAbort selectively proposing only aborted txns: txns:<start_ts:1 >
I0311 14:21:25.829625   16283 draft.go:928] abortOldTransactions for 1 txns. Error: <nil>
```

with `be=true` it hangs as well.

i think it's the case why we have all our problems for last time. nodes get stuck for all the time and i have to restart them manually because they answer `OK` for healthcheck. but all mutations and queries just run forever
it would be cool if you fix this issue in next release or at least some rc

**Comment by danielmai** [comment]
Can reproduce. A fix for this is in v1.0.14-rc1. Can you confirm that this is OK in that versiono @makitka2007?

**Comment by makitka2007** [comment]
i was a little bit confused about commit from this rc, cause it's related to "best effort" and we are not using it yet
but seems like fix works, dgraph is no more freezing after uncommitted transaction
thank you

**Comment by andrewsmedina** [comment]
@danielmai I have the same problem with my v1.0.13 production cluster. It's safe to use the 1.0.14-rc1 in production?

**Comment by danielmai** [comment]
We released v1.0.14-rc2 yesterday. These releases only contain bug fixes from v1.0.13. You can review the full changelists for each version here:

https://github.com/dgraph-io/dgraph/compare/v1.0.13...v1.0.14-rc1
https://github.com/dgraph-io/dgraph/compare/v1.0.13...v1.0.14-rc2
